### PR TITLE
Clean up the beam_utils module

### PR DESCRIPTION
**Pull request by bjorng**
The `beam_utils` module has functions such as `is_not_used/3` and `is_not_killed/3` that were used in by many optimization passes up to OTP 21.

The introduction of the SSA intermediate format eliminated all of the uses of those functions in `beam_utils`.

This pull request eliminates the last use major of uses of `beam_utils` and removes most functions from `beam_utils`.

The `beam_jump` pass no longer uses `beam_utils:is_killed/3` and in rare cases will produce slightly worse code.

The `beam_trim` pass no longer uses `beam_utils:is_not_used/3`, but it is now able to apply the stack trimming optimization in more circumstances than before.

For more details, see the individual commit messages.

### CODE REVIEW

File: lib/compiler/src/beam_trim.erl [hunk 1]
```diff
@@ -297,3 +330,76 @@ frame_size_branch(L, Is, Safe) ->
...
+    Y =:= Yreg orelse is_not_used(Y, Is);
+is_not_used(Y, [{line,_}|Is]) ->
+    is_not_used(Y, Is);
+is_not_used(Y, [{make_fun2,_,_,_,_}|Is]) ->
+    is_not_used(Y, Is);
+is_not_used(Y, [{test,_,_,Ss}|Is]) ->
+    not member(Y, Ss) andalso is_not_used(Y, Is);
+is_not_used(Y, [{test,Op,Fail,Live,Ss,Dst}|Is]) ->
+    Block = [{set,[Dst],Ss,{alloc,Live,{bif,Op,Fail}}}],
+    is_not_used(Y, [{block,Block}|Is]).
```
Comment (michalmuskala) [inline lib/compiler/src/beam_trim.erl]: Should this also use `is_not_used_ss_dst`?

File: lib/compiler/src/beam_trim.erl [hunk 2]
```diff
@@ -132,14 +136,14 @@ create_map(Trim, []) ->
        (Any) -> Any
     end;
 create_map(Trim, Moves) ->
-    GbTree0 = [{Src,Dst-Trim} || {move,{y,Src},{y,Dst}} <- Moves],
-    GbTree = gb_trees:from_orddict(sort(GbTree0)),
+    Map0 = [{Src,Dst-Trim} || {move,{y,Src},{y,Dst}} <- Moves],
+    Map = maps:from_list(Map0),
     IllegalTargets = gb_sets:from_list([Dst || {move,_,{y,Dst}} <- Moves]),
```
Comment (michalmuskala) [inline lib/compiler/src/beam_trim.erl]: Would it make sense to change this to `cerl_sets` as well?

File: lib/compiler/src/beam_trim.erl (hunk 1 above)
Comment (bjorng) [reply lib/compiler/src/beam_trim.erl]: Yes, it should.

File: lib/compiler/src/beam_trim.erl (hunk 2 above)
Comment (bjorng) [reply lib/compiler/src/beam_trim.erl]: Yes, it does.

**Pull request merged**